spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultPulsarProducerFactory.java

```diff
@@ -40,76 +43,88 @@
+	public Producer<T> createProducer(Schema<T> schema, @Nullable String topic) throws PulsarClientException {
+		return doCreateProducer(schema, topic, null, null);
 	}
 
 	@Override
-	public Producer<T> createProducer(@Nullable String topic, Schema<T> schema, @Nullable MessageRouter messageRouter,
-			@Nullable List<ProducerInterceptor> producerInterceptors,
-			@Nullable List<ProducerBuilderCustomizer<T>> producerBuilderCustomizers) throws PulsarClientException {
-		return doCreateProducer(topic, schema, messageRouter, producerInterceptors, producerBuilderCustomizers);
+	public Producer<T> createProducer(Schema<T> schema, @Nullable String topic,
+			@Nullable Collection<String> encryptionKeys, @Nullable List<ProducerBuilderCustomizer<T>> customizers)
+			throws PulsarClientException {
+		return doCreateProducer(schema, topic, encryptionKeys, customizers);
 	}
 
 	/**
 	 * Create the actual producer.
+	 * @param schema the schema of the messages to be sent
 	 * @param topic the topic the producer will send messages to or {@code null} to use
 	 * the default topic
-	 * @param schema the schema of the messages to be sent
-	 * @param messageRouter the optional message router to use
-	 * @param producerInterceptors the optional producer interceptors to use
-	 * @param producerBuilderCustomizers the optional list of customizers to apply to the
-	 * producer builder
+	 * @param encryptionKeys the encryption keys used by the producer or {@code null} to
+	 * use the default encryption keys
+	 * @param customizers the optional list of customizers to apply to the producer
+	 * builder
 	 * @return the created producer
 	 * @throws PulsarClientException if any error occurs
 	 */
-	protected Producer<T> doCreateProducer(@Nullable String topic, Schema<T> schema,
-			@Nullable MessageRouter messageRouter, @Nullable List<ProducerInterceptor> producerInterceptors,
-			@Nullable List<ProducerBuilderCustomizer<T>> producerBuilderCustomizers) throws PulsarClientException {
-		final String resolvedTopic = ProducerUtils.resolveTopicName(topic, this);
+	@SuppressWarnings("unchecked")
+	protected Producer<T> doCreateProducer(Schema<T> schema, @Nullable String topic,
+			@Nullable Collection<String> encryptionKeys, @Nullable List<ProducerBuilderCustomizer<T>> customizers)
+			throws PulsarClientException {
+		String resolvedTopic = ProducerUtils.resolveTopicName(topic, this);
 		this.logger.trace(() -> String.format("Creating producer for '%s' topic", resolvedTopic));
-		final ProducerBuilder<T> producerBuilder = this.pulsarClient.newProducer(schema);
-		if (!CollectionUtils.isEmpty(this.producerConfig)) {
-			producerBuilder.loadConf(this.producerConfig);
+		ProducerBuilder<T> producerBuilder = this.pulsarClient.newProducer(schema);
+
+		Map<String, Object> config = new HashMap<>(this.producerConfig);
+
+		if (encryptionKeys != null) {
+			config.put("encryptionKeys", encryptionKeys);
+		}
+
+		producerBuilder.loadConf(config);
```

Comment:
Since it's only used here, does it have to go to a util ?
I could move it to a private method for readability ?
Note that we need to pass the resolvedTopic (or include the call to `ProducerUtils.resolveTopicName` in the method)

Reply:
Agreed, private method in class is totally fine, it is really just readability that I am after. Ty